File: leet/blind75/buyStock1.js
function maxProfit(prices) {
  let [buy, sell] = [0,1]
  let max = 0
  while(sell < prices.length) {
    let profit = prices[sell] - prices[buy]
    if(profit < 0) {
      buy = sell
    }
    if(profit > max) {
      max = profit
    }
    sell++
  }
  return max
}


console.log(maxProfit([7,1,5,3,6,4]))

File: leet/blind75/buyStock1.py
def maxProfit(prices: list[int]) -> int:
    [buy, sell] = [0, 1]
    maxProf = 0
    while sell < len(prices):
        curr = prices[sell] - prices[buy]
        if curr < 0:
            buy = sell
        if curr > maxProf:
            maxProf = curr
        sell += 1
    return maxProf


print(maxProfit([7,1,5,3,6,4]))

